chevah/github_hooks_server/handler.py

```diff
@@ -14,25 +14,16 @@ class Handler(object):
     Should implement a method with the same name as github event.
     """
 
-    USERS_GITHUB_TO_TRAC = {
-        'adiroiban': 'adi',
-        'lgheorghiu': 'laura',
-        'hcs0': 'hcs',
-        }
-
     RE_TRAC_TICKET_ID = r'\[#(\d+)\] .*'
     RE_REVIEWERS = r'.*reviewers{0,1}:{0,1} @.*'
     RE_NEEDS_REVIEW = r'.*needs{0,1}[\-_]review.*'
     RE_NEEDS_CHANGES = r'.*needs{0,1}[\-_]changes{0,1}.*'
     RE_APPROVED = r'.*(changes{0,1}[\-_]approved{0,1})|(approved-at).*'
 
-    # Helper for tests.
-    _current_ticket = None
-
-    def __init__(self, trac_url, github_token):
-        self._github = github3.login(token=github_token)
+    def __init__(self, github):
+        self._github = github
         if not self._github:
-            raise RuntimeError('Failed to init GitHut.')
+            raise RuntimeError('Failed to init GitHub.')
```

Comment:
Just asking. Are all the exceptions logged by default?

Reply:
We have an all-exception handler in `hook` in `server.py`:

```python
except:
    import traceback
    logging.error(
        f'Failed to process "{event_name}":\n'
        f'{content}\n'
        f'{traceback.format_exc()}'
        )
    return func.HttpResponse(
        body="Error:003: Internal error", status_code=500
        )
```

But this particular error would be missed, since I see the Handler is created at top-level, not inside the hook method.

I will move all meaningful code inside the try block. So, the only uncaught (and unlogged) errors would be also visible offline.

Thank you for this observation!

Reply:
No problem.
There is also this option - https://stackoverflow.com/a/16993115/539264